tmc-langs-r/src/main/java/fi/helsinki/cs/tmc/langs/r/RPlugin.java

```
@@ -0,0 +1,158 @@
+
+    private static Logger log = LoggerFactory.getLogger(RPlugin.class);
+
+    public RPlugin() {
+        super(
+                new ExerciseBuilder(),
+                new StudentFileAwareSubmissionProcessor(),
+                new StudentFileAwareZipper(),
+                new StudentFileAwareUnzipper());
+    }
+
+    /*
+    NOTE: Files.exists does not seem to be able to verify the R and
+    testthat folder's existence if they are empty.
+    */
+    @Override
+    public boolean isExerciseTypeCorrect(Path path) {
+        return Files.exists(path.resolve(R_FOLDER_PATH))
+                || Files.exists(path.resolve(TEST_FOLDER_PATH).resolve(TESTTHAT_FOLDER_PATH))
+                || Files.exists(path.resolve(TEST_FOLDER_PATH).resolve(TESTTHAT_FILE_PATH));
+    }
+
+    @Override
+    protected StudentFilePolicy getStudentFilePolicy(Path projectPath) {
+        return new RStudentFilePolicy(projectPath);
+    }
+
+    @Override
+    public String getPluginName() {
+        return "r";
+    }
+
+    @Override
+    public Optional<ExerciseDesc> scanExercise(Path path, String exerciseName) {
+        ProcessRunner runner = new ProcessRunner(this.getAvailablePointsCommand(), path);
+
+        try {
+            runner.call();
+        } catch (Exception e) {
+            log.error(CANNOT_SCAN_EXERCISE_MESSAGE, e);
+        }
+
+        try {
+            ImmutableList<TestDesc> testDescs = new RExerciseDescParser(path).parse();
+            return Optional.of(new ExerciseDesc(exerciseName, testDescs));
+        } catch (IOException e) {
+            log.error(CANNOT_PARSE_EXERCISE_DESCRIPTION_MESSAGE, e);
+        }
+
+        return Optional.absent();
+    }
+
+    @Override
+    public RunResult runTests(Path path) {
+        ProcessRunner runner = new ProcessRunner(getTestCommand(), path);
+
+        try {
+            runner.call();
+        } catch (Exception e) {
+            log.error(CANNOT_RUN_TESTS_MESSAGE, e);
+        }
+
```

What happens if running the tests fails but there is already a results file in the directory? I'm worried about this failing silently and then showing the previous run's results to the user.